lib/fs-watch/src/lib.rs

```diff
@@ -812,4 +819,26 @@ mod tests {
     fn inotify_nonexistent_files_dont_file_delete_events() {
         Fixture::new().test_inotify(test_detects_delete_and_recreate)
     }
+
+    const LONG: &str = "loooooooooooooooooooooooooooooooooooooooooooooooongboi";
```

Comment:
this might be a good job for quickcheck

Reply:
Yeah, that's a good point. I'm going to go ahead and merge the bugfix first, though, so it's not blocked on writing a QuickCheck generator for valid filenames?